Lib/test/regrtest.py

```diff
@@ -544,10 +561,13 @@ def work():
                         output.put((None, None, None, None))
                         return
                     # -E is needed by some tests, e.g. test_import
-                    popen = Popen(base_cmd + ['--slaveargs', json.dumps(args_tuple)],
-                                   stdout=PIPE, stderr=PIPE,
-                                   universal_newlines=True,
-                                   close_fds=(os.name != 'nt'))
+                    args = base_cmd + ['--slaveargs', json.dumps(args_tuple)]
+                    if testdir:
+                        args.extend(('--testdir', testdir))
```

Comment:
> But master don't pass testdir too.

In master, there is no need to pass testdir since regrtest is designed differently. run_test_in_subprocess() pass "ns" a dict which contains testdir. In the slave, setup_tests() is called with ns and so sys.path is updated properly.

I hate regrtest.py of Python 2.7/3.5, I would prefer to just replace it with master libregrtest...